cocos/navmesh/CCNavMesh.cpp

```diff
@@ -32,20 +32,22 @@
 
 NS_CC_BEGIN
 
+#pragma pack(push,1)
 struct TileCacheSetHeader
 {
-    int magic;
-    int version;
-    int numTiles;
+    int32_t magic;
+    int32_t version;
+    int32_t numTiles;
```

Comment:
nope, that was the test case. Well, perhaps it runs Ok in your devices. But it crashed in all my devices: Pixel 5, Pixel 6, Nexus 5x and Galaxy S4.
I used GCC 4.9, NDK 13 and build tools 25.